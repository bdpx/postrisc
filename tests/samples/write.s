.rodata
    align	16
console_test_quad:
    quad	1.189731495357231765085759326628007e+4932
console_test_quad2:
    quad	1.23456789 + 32.0
console_test_quad3:
    quad	0.2345678901234567890123456789012345678 + 0.2
    quad	2*asin(1)
    quad	255
console_test_double:
    double	acos(sin(3.1415926)) ;-1.2345678e+200
    double	444.689679
console_test_float:
    float	0.123456789123456789e+30
    float	2.123456789122233
    float	0.0
    float	1.0
.text
    alloc	35
    write	"ip=%s(ip), eip=%s(eip), psr=%s(psr)"

    write	"end test write special regs"

    write	"\ntest write: general register"

    write	"%%i8(sp)  = %i8(sp)"
    write	"%%i16(sp) = %i16(sp)"
    write	"%%i32(sp) = %i32(sp)"
    write	"%%i64(sp) = %i64(sp)"
    write	"%%u8(sp)  = %u8(sp)"
    write	"%%u16(sp) = %u16(sp)"
    write	"%%u32(sp) = %u32(sp)"
    write	"%%u64(sp) = %u64(sp)"
    write	"%%x8(sp)  = 0x%x8(sp)"
    write	"%%x16(sp) = 0x%x16(sp)"
    write	"%%x32(sp) = 0x%x32(sp)"
    write	"%%x64(sp) = 0x%x64(sp)"

    write	"%x64(r0)"
    write	"%x64(r1)"
    write	"%x64(r2)"
    write	"%x64(r22)"
    write	"%x64(r33)"
    write	"%x64(g0)"
    write	"%x64(g1)"
    write	"%x64(tp)"
    write	"%x64(sp)"

    write	"end test write general regs"

    ld.q.r	%r22, console_test_quad
    write	"r22 = %x128(r22) %f128(r22)"
    ld.q.r	%r22, console_test_quad2
    write	"r22 = %x128(r22) %f128(r22)"
    ld.q.r	%r22, console_test_quad3
    write	"r22 = %x128(r22) %f128(r22)"
    ldz.d.r	%r22, console_test_double
    write	"r22 = %x64(r22) %f64(r22)"
    ldz.w.r	%r22, console_test_float
    write	"r22 = %x32(r22) %f32(r22)"

    write	"end test write fp regs"
.end
