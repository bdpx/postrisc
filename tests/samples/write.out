ip=fffffffff1230000, eip=fffffffff1230100, psr=0000000000020000 pm=0 id=0 dd=0 lp=0 tb=0 ss=0 ic=0 ib=0 pl=0 pp=0 ri=2 future=00000000000000000000000000000000
end test write special regs

test write: general register
%i8(sp)  = 0
%i16(sp) = 0
%i32(sp) = -245104640
%i64(sp) = -245104640
%u8(sp)  = 0
%u16(sp) = 0
%u32(sp) = 4049862656
%u64(sp) = 18446744073464446976
%x8(sp)  = 0x00
%x16(sp) = 0x0000
%x32(sp) = 0xf1640000
%x64(sp) = 0xfffffffff1640000
0000000000000000
0000000000000000
0000000000000000
0000000000000000
0000000000000000
0000000000000000
0000000000000000
fffffffff1830000
fffffffff1640000
end test write general regs
r22 = 7ffeffffffffffff_ffffffffffffffff +1.1897314953572317650857593266280070e+4932
r22 = 400409e0652141ef_0dbf5b4a289824a4 +3.3234567889999999999999999999999997e+01
r22 = 3ffdbcff5d6fe34b_a936bc527a7a8239 +4.3456789012345678901234567890123456e-01
r22 = 3ff921fb45e1837b 1.5707962732051033e+00
r22 = 6fc77488 1.2345679e+29
end test write fp regs
